### DISCUSSION TITLE
Default Browser option

### DISCUSSION THREAD

**Discussion by Mirthany** [U-C-S/Hurl#122] [Q&A]
Hi,

I have created several rules to open different browsers. Is there an option to create a rule or an option that applies when no other rules match? Like a default fall back.

**Comment by eidylon** [comment]
Oh, that's a good idea. But it should have a way to launch the selection window too. 
Maybe, if you have a default/fallback rule, then it will launch unmatched URLs with that. 
But if you launch a URL while holding the CTRL key, then it will launch the selection dialog.

**Reply by U-C-S** [reply]
> launch a URL while holding the CTRL key

Not sure how to detect and implement this behavior, But I personally want this for my use cases for a while. I will see about it soon enough.

**Reply by eidylon** [reply]
@U-C-S - I do this in some of my code, where if the user holds down CTRL while performing operations, then I log out extra debugging info. This is the way I do it.... in a static library class I have for utility functions....

```
using System.Windows.Forms;

static class Library {
    public static bool CtrlKeyOn => (Control.ModifierKeys & Keys.Control) != 0;
}
```

**Comment by Jay-o-Way** [comment]
Shouldn't bee too difficult, I think. Would be a setting. Something like
- search existing rules (apply when match)
- no match? check setting and:
  - open selection window
  - open default (OS) handler

**Comment by romanlex** [comment]
what about default browser ?

**Comment by pg-austinlogic** [comment]
You can create a rule to match any url at the end of the rule list like the following:

    {
      "Rules": [
        "r$[\\s\\S]*"
      ],
      "RulesetName": "Default",
      "BrowserName": "Microsoft Edge"
    }

**Reply by romanlex** [reply]
this regular can be as "Default" checkbox
